rules/otp.bzl

```diff
@@ -0,0 +1,150 @@
+        "OWNER_SW_CFG_DIGEST": "0x0",
+        "OWNER_SW_CFG_ROM_BOOTSTRAP_EN": "0x739",
+        "OWNER_SW_CFG_ROM_KEYMGR_ROM_EXT_MEAS_EN": "0x0",
+        "OWNER_SW_CFG_ROM_ALERT_CLASS_EN": "0xa9a9a9a9",
+        "OWNER_SW_CFG_ROM_ALERT_ESCALATION": "0xd1d1d1d1",
+        "OWNER_SW_CFG_ROM_ALERT_CLASSIFICATION": ["0x94949494"] * 80,
+        "OWNER_SW_CFG_ROM_LOCAL_ALERT_CLASSIFICATION": ["0x94949494"] * 16,
+        "OWNER_SW_CFG_ROM_ALERT_ACCUM_THRESH": ["0x00000000"] * 4,
+        "OWNER_SW_CFG_ROM_ALERT_TIMEOUT_CYCLES": ["0x00000000"] * 4,
+        "OWNER_SW_CFG_ROM_ALERT_PHASE_CYCLES": [
+            "0x0",
+            "0xa",
+            "0xa",
+            "0xFFFFFFFF",
+            "0x0",
+            "0xa",
+            "0xa",
+            "0xFFFFFFFF",
+            "0x0",
+            "0x0",
+            "0x0",
+            "0x0",
+            "0x0",
+            "0x0",
+            "0x0",
+            "0x0",
+        ],
+        "OWNER_SW_CFG_ROM_ALERT_DIGEST_RMA": "0x36ed9cb0",
+    }
+
+    # HW_CFG
+    hw_cfg_items = {
+        "DEVICE_ID": "<random>",
+        "EN_CSRNG_SW_APP_READ": "true",
+        "EN_ENTROPY_SRC_FW_READ": "true",
+        "EN_ENTROPY_SRC_FW_OVER": "true",
+    }
+
+    # SECRET0
+    secret0_items = {
+        "TEST_UNLOCK_TOKEN": "<random>",
+        "TEST_EXIT_TOKEN": "<random>",
+    }
+
+    # SECRET1
+    secret1_items = {
+        "FLASH_ADDR_KEY_SEED": "<random>",
+        "FLASH_DATA_KEY_SEED": "<random>",
+        "SRAM_DATA_KEY_SEED": "<random>",
+    }
+
+    # SECRET2
+    secret2_items = {
+        "RMA_TOKEN": "<random>",
+        "CREATOR_ROOT_KEY_SHARE0": "<random>",
+        "CREATOR_ROOT_KEY_SHARE1": "<random>",
+    }
+
+    # Assemble all OTP paritions
+    otp["partitions"] = [
+        {
+            "name": "CREATOR_SW_CFG",
+            "items": _expand_partition_items(creator_sw_cfg_items),
+        },
+        {
+            "name": "OWNER_SW_CFG",
+            "items": _expand_partition_items(owner_sw_cfg_items),
+        },
+        {
+            "name": "HW_CFG",
+            "lock": "True",
+            "items": _expand_partition_items(hw_cfg_items),
+        },
+        {
+            "name": "SECRET0",
+            "lock": "True",
+            "items": _expand_partition_items(secret0_items),
+        },
+        {
+            "name": "SECRET1",
+            "lock": "True",
+            "items": _expand_partition_items(secret1_items),
+        },
+        {
+            "name": "SECRET2",
+            "lock": "False",
+            "items": _expand_partition_items(secret2_items),
+        },
+        {
+            "name": "LIFE_CYCLE",
+            "state": ctx.attr.lc_state,
+            "count": ctx.attr.lc_count,
+        },
+    ]
+
+    file = ctx.actions.declare_file("{}.json".format(ctx.attr.name))
+    ctx.actions.write(file, json.encode_indent(otp))
+    return DefaultInfo(
+        files = depset([file]),
+        runfiles = ctx.runfiles(files = [file]),
```

Comment:
Does it make sense to use `runfiles` when this rule is not executable? I think it can be omitted here, but I'm not 100% confident.